### PR TITLE
fix(dlc): prevent sentinel detection forgery in pr-check

### LINKED ISSUE #243: fix(dlc): pr-check sentinel detection is forgeable via raw substring scan
## Context

While fixing #173, reviewers found a pre-existing gap in `dlc:pr-check`'s "already replied" detection: it's vulnerable to sentinel forgery by any reviewer, and the resulting silent-skip is invisible to the coverage-verification safety net.

`plugins/dlc/scripts/pr-comments.sh:368`:

```jq
select(.author != $pr_author and (.body | contains("<!-- dlc-reply:") | not))
```

This is a **raw substring scan** on the comment's `body` field — not an ID match, not scoped to DLC's own comments. Any reviewer (not just the PR author) can include the literal substring `<!-- dlc-reply:` in their own comment. That comment is then filtered out of `$reviewer_issue_comments` entirely — `pr-check` never sees it, never categorizes it, and never replies to it.

## Why this is worse than a missed reply

`SKILL.md` Step 4b (Coverage Verification) computes its coverage targets from the same filtered inventory this jq predicate produces. A comment excluded here is excluded from Step 4b's expected counts too — so the FATAL "coverage verification failed" gate, which exists specifically to catch silently-dropped comments, **cannot catch this one**. It's a silent miss by construction, not just in practice.

A related but distinct risk: if the forged substring includes a fake `{database_id}` that happens to match a *different*, real comment's ID (e.g. `<!-- dlc-reply:12345 -->` where 12345 is another comment's actual ID), that other comment could also be misclassified as "already replied" on a future run, even though DLC never actually replied to it.

## Scope Note

This predates #173 and is not something that PR's excerpt-sanitization changes touch — fencing/blockquoting/`@`-stripping the *quoted excerpt* in DLC's own replies doesn't affect this jq filter, which scans the *original* reviewer comment's raw body directly via the GitHub API response, before DLC ever composes anything.

## Proposed Fix

Options to evaluate (not prescribing one — needs design thought):
- Require the sentinel to appear in a comment **authored by the DLC actor/bot account**, not just anywhere in any comment's body (closes the forgery vector directly)
- Switch "already replied" detection to a stronger anchor than a raw substring — e.g. require the sentinel plus author check, or track replied-IDs out-of-band instead of scanning body text
- At minimum, scope the `contains()` check to `.author == $dlc_actor_login` if that's available from the API response

## Acceptance Criteria

- [ ] A reviewer comment containing the literal substring `<!-- dlc-reply:` cannot cause itself (or another comment) to be misclassified as already-replied
- [ ] Coverage verification (Step 4b) still catches any comment that `pr-check` fails to categorize, including ones affected by this forgery vector

## Implementation Hints

- File: `plugins/dlc/scripts/pr-comments.sh:368` (the vulnerable predicate)
- File: `plugins/dlc/skills/pr-check/SKILL.md` (Step 4b coverage-verification logic, which derives its targets from the same filtered inventory)

## Related

- Follow-up from #173

### PR DESCRIPTION
**Pull request by rube-de**
## Summary

Fixes #243

Hardens `dlc:pr-check` and related scripts against sentinel detection forgery via raw substring scans.

### Context & Root Cause
Previously, `pr-comments.sh` and `fetch-merged-pr-comments.sh` used unauthenticated raw substring scans on comment bodies (`select((.body | contains("<!-- dlc-reply:") | not))`) to exclude DLC reply sentinels from reviewer inventories. Any reviewer mentioning `<!-- dlc-reply:` or inserting a forged sentinel in an issue comment would cause their comment to be filtered out of `reviewer_issue_comments` and reviewer counts, silently evading Step 4b coverage verification. Furthermore, unstructured scans for sentinels in `ISSUE_COMMENTS` could allow a forged database ID to mark another reviewer's comment as already replied.

### Changes
1. **Query authenticated viewer in `pr-comments.sh`**: Added `viewer { login }` to the GraphQL query to identify the executing actor and included `viewer` in the returned `pr` JSON object.
2. **Authenticated sentinel filtering**: Replaced raw substring filters in `pr-comments.sh` and `fetch-merged-pr-comments.sh` with structured regex and actor-scoped checks (`test("<!--\\s*dlc-reply:[0-9]+\\s*-->")` scoped to ``, ``, or bot accounts). Reviewer comments mentioning or containing the sentinel string remain in reviewer inventory.
3. **Updated `pr-check/SKILL.md`**: Explicitly documented that "already replied" detection in Step 2 requires the sentinel to be authored by `PR_AUTHOR`, `PR_VIEWER`, or an automated bot account (and created after the target comment for issue comments), neutralizing cross-reviewer forgery.
4. **Documentation**: Updated `docs/learnings.md` with the hardened sentinel attribution pattern.

### Verification
- Tested `pr-comments.sh` live against PR #256; confirmed `.pr.viewer` and structured output.
- Validated with `jq` test suites covering all edge cases (PR author comments, viewer sentinel replies, viewer normal comments, reviewer comments with sentinels, reviewer comments mentioning sentinels, bot reviewer comments).
- Verified POSIX shell script syntax using `sh -n`.

## Summary by CodeRabbit

- **Bug Fixes**
  - Improved pull request comment handling to distinguish authentic automated replies from reviewer comments.
  - Prevented comment markers in unrelated reviewer comments from being misinterpreted as replies.
  - Improved recognition of replies from CI and automation accounts.
  - Made “already replied” checks more accurate by validating authorship, markers, and comment order.
  - Added clearer handling for non-actionable comments and authentication lookup failures.

- **Documentation**
  - Updated guidance and references for reliable reply detection.

### CODE REVIEW

File: plugins/dlc/scripts/pr-comments.sh [hunk 1]
```diff
@@ -362,10 +363,18 @@ jq --arg owner "$OWNER" --arg repo "$REPO" '
...
+  # (authored by PR author, viewer actor, or bot with a valid dlc-reply sentinel).
+  # Both are retained in the raw $issue_comments array for "already replied" detection.
+  # Reviewer comments mentioning "<!-- dlc-reply:" are NOT excluded.
   [ $issue_comments[] |
-    select(.author != $pr_author and (.body | contains("<!-- dlc-reply:") | not))
+    select(
+      .author != $pr_author and
+      (
+        ((.author == $viewer or (.author | test("(\\[bot\\]$|^github-actions$)"))) and
+         (.body | test("<!--\\s*dlc-reply:[0-9]+\\s*-->"))) | not
```
[the remark this answers is not in the document]
Comment (rube-de) [reply plugins/dlc/scripts/pr-comments.sh]: Fixed: simplified sentinel filtering in pr-comments.sh to strictly check .author == $viewer rather than broadly trusting all bot accounts

File: plugins/dlc/scripts/fetch-merged-pr-comments.sh [hunk 2]
```diff
@@ -385,17 +380,22 @@ while [ "$_idx" -lt "$_pr_count" ]; do
...
+    # sentinels (from bots/actors) — neither carries reviewer signal.
     [ $pr.comments.nodes[] |
       select(.body != null and (.body | gsub("\\s"; "") | length > 0)) |
       select((.author.login // "ghost") != $pr_author) |
-      select(.body | contains("<!-- dlc-reply:") | not) |
+      select(
+        (
+          ((.author.__typename // "") == "Bot" or ((.author.login // "") | test("(\\[bot\\]$|^github-actions$)"))) and
+          ((.body // "") | test("<!--\\s*dlc-reply:[0-9]+\\s*-->"))
+        ) | not
```
[the remark this answers is not in the document]
Comment (rube-de) [reply plugins/dlc/scripts/fetch-merged-pr-comments.sh]: Fixed: updated fetch-merged-pr-comments.sh to filter authentic trailing sentinel replies on issue comments regardless of author account type

File: plugins/dlc/scripts/fetch-merged-pr-comments.sh [hunk 3]
```diff
@@ -343,18 +343,16 @@ while [ "$_idx" -lt "$_pr_count" ]; do
...
       . as $c |
       select(($c.author.login // "ghost") != $pr_author) |
-      select(($c.body // "") | contains("<!-- dlc-reply:") | not) |
       {
         id:                 ($c.id // $thread.id),
         type:               "thread",
         author:             ($c.author.login // "ghost"),
-        is_bot:             (($c.author.__typename // "") == "Bot"),
+        is_bot:             (($c.author.__typename // "") == "Bot" or (($c.author.login // "") | endswith("[bot]"))),
         body:               (($c.body // "") | .[0:2000]),
```
[the remark this answers is not in the document]
Comment (rube-de) [reply plugins/dlc/scripts/fetch-merged-pr-comments.sh:372]: Fixed: aligned is_bot computation across threads, review bodies, and issue comments to consistently recognize bot accounts and github-actions

File: plugins/dlc/scripts/fetch-merged-pr-comments.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (rube-de) [reply plugins/dlc/scripts/fetch-merged-pr-comments.sh]: Fixed: updated fetch-merged-pr-comments.sh to filter authentic trailing sentinel replies across human maintainers and bots alike

File: plugins/dlc/scripts/pr-comments.sh [hunk 4]
```diff
@@ -362,10 +363,15 @@ jq --arg owner "$OWNER" --arg repo "$REPO" '
...
+  # exclude PR author comments and authentic DLC sentinel replies
+  # (authored by viewer actor with a valid dlc-reply sentinel).
+  # Both are retained in the raw $issue_comments array for "already replied" detection.
+  # Reviewer comments mentioning "<!-- dlc-reply:" are NOT excluded.
   [ $issue_comments[] |
-    select(.author != $pr_author and (.body | contains("<!-- dlc-reply:") | not))
+    select(
+      .author != $pr_author and
+      ((.author == $viewer and (.body | test("<!--\\s*dlc-reply:[0-9]+\\s*-->"))) | not)
+    )
```
[the remark this answers is not in the document]
Comment (rube-de) [reply plugins/dlc/scripts/pr-comments.sh:387]: Fixed: included github-actions[bot] and github-actions CI actors alongside viewer actor in pr-comments.sh sentinel filter

File: plugins/dlc/scripts/fetch-merged-pr-comments.sh [hunk 5]
```diff
@@ -385,17 +380,17 @@ while [ "$_idx" -lt "$_pr_count" ]; do
...
 
-    # Issue comments → comments[]. Exclude PR-author comments and DLC reply
+    # Issue comments → comments[]. Exclude PR-author comments and authentic DLC reply
     # sentinels — neither carries reviewer signal.
     [ $pr.comments.nodes[] |
       select(.body != null and (.body | gsub("\\s"; "") | length > 0)) |
       select((.author.login // "ghost") != $pr_author) |
-      select(.body | contains("<!-- dlc-reply:") | not) |
+      select(((.body // "") | test("(^|\\n)<!--\\s*dlc-reply:[0-9]+\\s*-->\\s*$")) | not) |
       {
```
[the remark this answers is not in the document]
Comment (rube-de) [reply plugins/dlc/scripts/fetch-merged-pr-comments.sh:405]: Fixed: scoped sentinel exclusion in fetch-merged-pr-comments.sh to CI bot accounts to prevent reviewer comment drops via forged trailing sentinels

File: plugins/dlc/skills/pr-check/SKILL.md [hunk 6]
```diff
@@ -178,28 +178,27 @@ Using the `REVIEW_BODIES` array from Step 1, classify each review body.
 
 A review body is **Resolved** when any of these apply:
 
-1. **Already replied** — a DLC reply was posted for this review body (sentinel `<!-- dlc-reply:{database_id} -->` is present in `ISSUE_COMMENTS`).
+1. **Already replied** — a DLC reply was posted for this review body. A valid DLC reply is an issue comment in `ISSUE_COMMENTS` that: (a) is authored by `PR_AUTHOR` or `PR_VIEWER` (never a third-party reviewer), and (b) contains the sentinel `<!-- dlc-reply:{database_id} -->` matching this review body's `database_id`.
```
[the remark this answers is not in the document]
Comment (rube-de) [reply plugins/dlc/skills/pr-check/SKILL.md]: Fixed: updated SKILL.md and docs/learnings.md to consistently include github-actions[bot] as a valid DLC sentinel author

File: plugins/dlc/skills/pr-check/SKILL.md [hunk 7]
```diff
@@ -178,28 +178,27 @@ Using the `REVIEW_BODIES` array from Step 1, classify each review body.
...
 
-> **Note:** Every review body requires body inspection, not just `APPROVED` ones. A `COMMENTED` review whose body is "No actionable issues found" is Resolved — the `COMMENTED` state alone does not imply actionable content. DLC replies to review bodies are posted as issue comments (via `gh pr comment`), so "already replied" detection must scan `ISSUE_COMMENTS` for the sentinel — not the review body's own data.
+> **Note:** Every review body requires body inspection, not just `APPROVED` ones. A `COMMENTED` review whose body is "No actionable issues found" is Resolved — the `COMMENTED` state alone does not imply actionable content. DLC replies to review bodies are posted as issue comments (via `gh pr comment`), so "already replied" detection must scan `ISSUE_COMMENTS` for authentic sentinels authored by `PR_AUTHOR`, `PR_VIEWER`, or `github-actions[bot]` — not the review body's own data, and not comments authored by third-party reviewers.
```
[the remark this answers is not in the document]
Comment (rube-de) [reply plugins/dlc/skills/pr-check/SKILL.md]: Fixed: updated SKILL.md to explicitly allow both github-actions[bot] and github-actions CI actor logins across all sentinel detection rules

File: plugins/dlc/skills/pr-check/SKILL.md [hunk 8]
```diff
@@ -178,28 +178,27 @@ Using the `REVIEW_BODIES` array from Step 1, classify each review body.
...
->
+> **Note:** Issue comments have no `path`/`line` like threads, no `state` like review bodies, and no parent-child links (they are a flat array). To reliably detect prior DLC replies, check for the `<!-- dlc-reply:{database_id} -->` sentinel in issue comments authored by `PR_AUTHOR`, `PR_VIEWER`, or `github-actions[bot]` (third-party reviewer comments containing the literal sentinel text cannot forge a reply or resolve another reviewer's comment). Parse the body for actionable items (specific change requests, code findings, questions). If the body is purely informational (status updates, CI results with no action items) and does not require any follow-up, classify it as **Resolved**, even if no DLC reply was posted.
```
[the remark this answers is not in the document]
Comment (rube-de) [reply plugins/dlc/skills/pr-check/SKILL.md]: Fixed: restored subsequent comment creation-order requirement in SKILL.md issue-comment sentinel detection rules

File: plugins/dlc/scripts/pr-comments.sh [hunk 9]
```diff
@@ -362,10 +363,16 @@ jq --arg owner "$OWNER" --arg repo "$REPO" '
...
+  # exclude PR author comments and authentic DLC sentinel replies
+  # (authored by viewer actor or CI bot with a valid dlc-reply sentinel).
+  # Both are retained in the raw $issue_comments array for "already replied" detection.
+  # Reviewer comments mentioning "<!-- dlc-reply:" are NOT excluded.
   [ $issue_comments[] |
-    select(.author != $pr_author and (.body | contains("<!-- dlc-reply:") | not))
+    select(
+      .author != $pr_author and
+      (((.author == $viewer or .author == "github-actions[bot]" or .author == "github-actions") and
+        (.body | test("<!--\\s*dlc-reply:[0-9]+\\s*-->"))) | not)
```
[the remark this answers is not in the document]
Comment (rube-de) [reply plugins/dlc/scripts/pr-comments.sh]: Fixed: updated sentinel regex in pr-comments.sh to trailing-sentinel pattern matching to avoid false positives on comments that merely quote the sentinel

File: plugins/dlc/scripts/fetch-merged-pr-comments.sh [hunk 10]
```diff
@@ -130,6 +130,7 @@ else
   REPO=$(printf '%s\n' "$_repo_json" | jq -r '.name')
 fi
 [ -n "$OWNER" ] && [ -n "$REPO" ] || die_json "Could not parse owner/repo" "REPO_PARSE"
+VIEWER=$(gh api graphql -f query='query { viewer { login } }' -q .data.viewer.login 2>/dev/null || echo "")
```
[the remark this answers is not in the document]
Comment (rube-de) [reply plugins/dlc/scripts/fetch-merged-pr-comments.sh:148]: Fixed: added fail-fast validation on VIEWER lookup and non-empty viewer guard in fetch-merged-pr-comments.sh

File: plugins/dlc/scripts/fetch-merged-pr-comments.sh [hunk 11]
```diff
@@ -385,17 +382,22 @@ while [ "$_idx" -lt "$_pr_count" ]; do
...
     [ $pr.comments.nodes[] |
       select(.body != null and (.body | gsub("\\s"; "") | length > 0)) |
       select((.author.login // "ghost") != $pr_author) |
-      select(.body | contains("<!-- dlc-reply:") | not) |
+      select(
+        (
+          (($viewer != "" and (.author.login // "") == $viewer) or ((.author.login // "") | test("(\\[bot\\]$|^github-actions$)"))) and
+          ((.body // "") | test("(^|\\n)<!--\\s*dlc-reply:[0-9]+\\s*-->\\s*$"))
+        ) | not
+      ) |
```
[the remark this answers is not in the document]
Comment (rube-de) [reply plugins/dlc/scripts/fetch-merged-pr-comments.sh]: Fixed: narrowed sentinel exclusion in fetch-merged-pr-comments.sh strictly to viewer actor and github-actions CI accounts

File: plugins/dlc/scripts/fetch-merged-pr-comments.sh [hunk 12]
```diff
@@ -385,17 +382,22 @@ while [ "$_idx" -lt "$_pr_count" ]; do
...
       {
         id:                 .id,
         type:               "issue_comment",
         author:             (.author.login // "ghost"),
-        is_bot:             ((.author.__typename // "") == "Bot"),
+        is_bot:             ((.author.__typename // "") == "Bot" or ((.author.login // "") | test("(\\[bot\\]$|^github-actions$)"))),
```
[the remark this answers is not in the document]
Comment (rube-de) [reply plugins/dlc/scripts/fetch-merged-pr-comments.sh:409]: Fixed: restricted sentinel exclusion to viewer actor and explicit CI allowlist (github-actions[bot] and github-actions)

File: plugins/dlc/scripts/pr-comments.sh [hunk 13]
```diff
@@ -301,10 +302,10 @@ done
 
 jq --arg owner "$OWNER" --arg repo "$REPO" '
   .data.repository.pullRequest as $pr |
+  (.data.viewer.login // "unknown") as $viewer |
 
   # Extract PR author for has_author_reply detection
   ($pr.author.login // "unknown") as $pr_author |
```
[the remark this answers is not in the document]
Comment (rube-de) [reply plugins/dlc/scripts/pr-comments.sh]: Fixed: separated viewer query from PR GraphQL query and added fail-fast validation on VIEWER lookup in pr-comments.sh

File: plugins/dlc/skills/pr-check/SKILL.md [hunk 14]
```diff
@@ -178,28 +178,27 @@ Using the `REVIEW_BODIES` array from Step 1, classify each review body.
 
 A review body is **Resolved** when any of these apply:
 
-1. **Already replied** — a DLC reply was posted for this review body (sentinel `<!-- dlc-reply:{database_id} -->` is present in `ISSUE_COMMENTS`).
+1. **Already replied** — a DLC reply was posted for this review body. A valid DLC reply is an issue comment in `ISSUE_COMMENTS` that: (a) is authored by `PR_AUTHOR`, `PR_VIEWER`, or the CI actor (`github-actions[bot]` / `github-actions`) (never a third-party reviewer), and (b) contains the sentinel `<!-- dlc-reply:{database_id} -->` matching this review body's `database_id`.
```
[the remark this answers is not in the document]
Comment (rube-de) [reply plugins/dlc/skills/pr-check/SKILL.md]: Fixed: specified standalone trailing sentinel requirement across all resolution rules in SKILL.md

File: plugins/dlc/skills/pr-check/SKILL.md [hunk 15]
```diff
@@ -178,28 +178,27 @@ Using the `REVIEW_BODIES` array from Step 1, classify each review body.
...
 
-> **Note:** Every review body requires body inspection, not just `APPROVED` ones. A `COMMENTED` review whose body is "No actionable issues found" is Resolved — the `COMMENTED` state alone does not imply actionable content. DLC replies to review bodies are posted as issue comments (via `gh pr comment`), so "already replied" detection must scan `ISSUE_COMMENTS` for the sentinel — not the review body's own data.
+> **Note:** Every review body requires body inspection, not just `APPROVED` ones. A `COMMENTED` review whose body is "No actionable issues found" is Resolved — the `COMMENTED` state alone does not imply actionable content. DLC replies to review bodies are posted as issue comments (via `gh pr comment`), so "already replied" detection must scan `ISSUE_COMMENTS` for authentic sentinels authored by `PR_AUTHOR`, `PR_VIEWER`, or the CI actor (`github-actions[bot]` / `github-actions`) — not the review body's own data, and not comments authored by third-party reviewers.
```
[the remark this answers is not in the document]
Comment (rube-de) [reply plugins/dlc/skills/pr-check/SKILL.md]: Fixed: replaced blank line in blockquote with continuation marker to resolve markdownlint MD028

File: plugins/dlc/scripts/fetch-merged-pr-comments.sh [hunk 16]
```diff
@@ -385,17 +382,22 @@ while [ "$_idx" -lt "$_pr_count" ]; do
...
-    # sentinels — neither carries reviewer signal.
+    # Issue comments → comments[]. Exclude PR-author comments and authentic DLC reply
+    # sentinels (authored by executing viewer or CI bot) — neither carries reviewer signal.
     [ $pr.comments.nodes[] |
       select(.body != null and (.body | gsub("\\s"; "") | length > 0)) |
       select((.author.login // "ghost") != $pr_author) |
-      select(.body | contains("<!-- dlc-reply:") | not) |
+      select(
+        (
+          (($viewer != "" and (.author.login // "") == $viewer) or ((.author.login // "") | test("(\\[bot\\]$|^github-actions$)"))) and
```
[the remark this answers is not in the document]
Comment (rube-de) [reply plugins/dlc/scripts/fetch-merged-pr-comments.sh]: Fixed: narrowed sentinel exclusion in fetch-merged-pr-comments.sh to explicit CI actors (github-actions[bot] / github-actions) and viewer

File: plugins/dlc/skills/pr-check/SKILL.md (hunk 14 above)
[the remark this answers is not in the document]
Comment (rube-de) [reply plugins/dlc/skills/pr-check/SKILL.md]: Fixed: untrusted PR_AUTHOR from sentinel author allowlist in SKILL.md; only PR_VIEWER and CI actors are accepted

File: plugins/dlc/scripts/pr-comments.sh [hunk 17]
```diff
@@ -74,6 +74,7 @@ MAX_PAGES=20
 # Initial query — includes pageInfo for cursor-based pagination
 QUERY='
 query($owner: String!, $repo: String!, $number: Int!) {
+  viewer { login }
```
[the remark this answers is not in the document]
Comment (rube-de) [reply plugins/dlc/scripts/pr-comments.sh]: Fixed: separated viewer lookup into independent GraphQL query to avoid App token non-nullable User! null-propagation failures

File: plugins/dlc/scripts/pr-comments.sh [hunk 18]
```diff
@@ -301,10 +302,10 @@ done
 
 jq --arg owner "$OWNER" --arg repo "$REPO" '
   .data.repository.pullRequest as $pr |
+  (.data.viewer.login // "unknown") as $viewer |
```
[the remark this answers is not in the document]
Comment (rube-de) [reply plugins/dlc/scripts/pr-comments.sh]: Fixed: added fail-fast validation on VIEWER detection in pr-comments.sh matching fetch-merged-pr-comments.sh

File: plugins/dlc/scripts/fetch-merged-pr-comments.sh [hunk 19]
```diff
@@ -130,6 +130,8 @@ else
   REPO=$(printf '%s\n' "$_repo_json" | jq -r '.name')
 fi
 [ -n "$OWNER" ] && [ -n "$REPO" ] || die_json "Could not parse owner/repo" "REPO_PARSE"
+VIEWER=$(gh api graphql -f query='query { viewer { login } }' -q .data.viewer.login 2>/dev/null) || die_json "Failed to fetch authenticated viewer login" "VIEWER_FETCH"
```
[the remark this answers is not in the document]
Comment (rube-de) [reply plugins/dlc/scripts/fetch-merged-pr-comments.sh]: Fixed: captured gh stderr on viewer query and included error diagnostic in VIEWER_FETCH failure message

File: plugins/dlc/skills/pr-check/SKILL.md [hunk 20]
```diff
@@ -178,28 +178,27 @@ Using the `REVIEW_BODIES` array from Step 1, classify each review body.
 
 A review body is **Resolved** when any of these apply:
 
-1. **Already replied** — a DLC reply was posted for this review body (sentinel `<!-- dlc-reply:{database_id} -->` is present in `ISSUE_COMMENTS`).
+1. **Already replied** — a DLC reply was posted for this review body. A valid DLC reply is an issue comment in `ISSUE_COMMENTS` that: (a) is authored by `PR_VIEWER` or the CI actor (`github-actions[bot]` / `github-actions`) (never `PR_AUTHOR` or a third-party reviewer), and (b) contains a standalone trailing sentinel `<!-- dlc-reply:{database_id} -->` matching this review body's `database_id`.
```
[the remark this answers is not in the document]
Comment (rube-de) [reply plugins/dlc/skills/pr-check/SKILL.md]: Fixed: clarified in SKILL.md that PR_VIEWER may equal PR_AUTHOR when run locally by the PR author and is explicitly allowed

File: plugins/dlc/scripts/pr-comments.sh [hunk 21]
```diff
@@ -362,10 +367,18 @@ jq --arg owner "$OWNER" --arg repo "$REPO" '
...
+  # exclude PR author comments and authentic DLC sentinel replies
+  # (authored by viewer actor or CI bot with a valid dlc-reply sentinel).
+  # Both are retained in the raw $issue_comments array for "already replied" detection.
+  # Reviewer comments mentioning "<!-- dlc-reply:" are NOT excluded.
   [ $issue_comments[] |
-    select(.author != $pr_author and (.body | contains("<!-- dlc-reply:") | not))
+    select(
+      .author != $pr_author and
+      (
+        ((($viewer != "" and .author == $viewer) or .author == "github-actions[bot]" or .author == "github-actions") and
```
[the remark this answers is not in the document]
Comment (rube-de) [reply plugins/dlc/scripts/pr-comments.sh]: Fixed: extracted is_authentic_dlc_reply helper to unify sentinel-authenticity and allowlist logic across scripts

File: plugins/dlc/scripts/fetch-merged-pr-comments.sh [hunk 22]
```diff
@@ -385,17 +387,22 @@ while [ "$_idx" -lt "$_pr_count" ]; do
...
-    # sentinels — neither carries reviewer signal.
+    # Issue comments → comments[]. Exclude PR-author comments and authentic DLC reply
+    # sentinels (authored by executing viewer or CI bot) — neither carries reviewer signal.
     [ $pr.comments.nodes[] |
       select(.body != null and (.body | gsub("\\s"; "") | length > 0)) |
       select((.author.login // "ghost") != $pr_author) |
-      select(.body | contains("<!-- dlc-reply:") | not) |
+      select(
+        (
+          ((($viewer != "" and (.author.login // "") == $viewer) or (.author.login // "") == "github-actions[bot]" or (.author.login // "") == "github-actions") and
```
[the remark this answers is not in the document]
Comment (rube-de) [reply plugins/dlc/scripts/fetch-merged-pr-comments.sh]: Fixed: extracted shared is_authentic_dlc_reply helper to standardize sentinel check across scripts

File: plugins/dlc/scripts/pr-comments.sh [hunk 23]
```diff
@@ -65,6 +65,13 @@ fi
 _tmpdir=$(mktemp -d "${TMPDIR:-/tmp}/pr-comments.XXXXXX") || die_json "Failed to create temporary directory" "TMPDIR_CREATE"
 trap 'rm -rf "$_tmpdir"' EXIT
 
+# --- viewer detection ------------------------------------------------------
+
+if ! VIEWER=$(gh api graphql -f query='query { viewer { login } }' -q .data.viewer.login 2>"$_tmpdir/viewer_err.txt"); then
+  die_json "Failed to fetch authenticated viewer login: $(tr '"' "'" < "$_tmpdir/viewer_err.txt")" "VIEWER_FETCH"
```
[the remark this answers is not in the document]
Comment (rube-de) [reply plugins/dlc/scripts/pr-comments.sh]: Fixed: extracted viewer detection into fetch_viewer helper function in pr-comments.sh

File: plugins/dlc/scripts/fetch-merged-pr-comments.sh [hunk 24]
```diff
@@ -131,17 +131,24 @@ else
...
 
+# --- temp files ------------------------------------------------------------
+
+_tmpdir=$(mktemp -d "${TMPDIR:-/tmp}/fetch-merged-pr-comments.XXXXXX") || die_json "Failed to create temporary directory" "TMPDIR_CREATE"
+trap 'rm -rf "$_tmpdir"' EXIT
+
+# --- viewer detection ------------------------------------------------------
+
+if ! VIEWER=$(gh api graphql -f query='query { viewer { login } }' -q .data.viewer.login 2>"$_tmpdir/viewer_err.txt"); then
+  die_json "Failed to fetch authenticated viewer login: $(tr '"' "'" < "$_tmpdir/viewer_err.txt")" "VIEWER_FETCH"
```
[the remark this answers is not in the document]
Comment (rube-de) [reply plugins/dlc/scripts/fetch-merged-pr-comments.sh]: Fixed: extracted viewer detection into fetch_viewer helper function in fetch-merged-pr-comments.sh

File: plugins/dlc/scripts/pr-comments.sh [hunk 25]
```diff
@@ -65,6 +65,16 @@ fi
...
 trap 'rm -rf "$_tmpdir"' EXIT
 
+# --- viewer detection ------------------------------------------------------
+
+fetch_viewer() {
+  if ! VIEWER=$(gh api graphql -f query='query { viewer { login } }' -q .data.viewer.login 2>"$_tmpdir/viewer_err.txt"); then
+    die_json "Failed to fetch authenticated viewer login: $(tr '"' "'" < "$_tmpdir/viewer_err.txt")" "VIEWER_FETCH"
+  fi
+  [ -n "$VIEWER" ] || die_json "Authenticated viewer login is empty" "VIEWER_EMPTY"
+}
```
[the remark this answers is not in the document]
Comment (rube-de) [reply plugins/dlc/scripts/pr-comments.sh:77]: Fixed: updated fetch_viewer in pr-comments.sh to use // empty jq fallback and explicitly guard against null string

File: plugins/dlc/scripts/fetch-merged-pr-comments.sh [hunk 26]
```diff
@@ -131,17 +131,26 @@ else
...
+_tmpdir=$(mktemp -d "${TMPDIR:-/tmp}/fetch-merged-pr-comments.XXXXXX") || die_json "Failed to create temporary directory" "TMPDIR_CREATE"
+trap 'rm -rf "$_tmpdir"' EXIT
+# --- viewer detection ------------------------------------------------------
+
+fetch_viewer() {
+  if ! VIEWER=$(gh api graphql -f query='query { viewer { login } }' -q .data.viewer.login 2>"$_tmpdir/viewer_err.txt"); then
+    die_json "Failed to fetch authenticated viewer login: $(tr '"' "'" < "$_tmpdir/viewer_err.txt")" "VIEWER_FETCH"
+  fi
+  [ -n "$VIEWER" ] || die_json "Authenticated viewer login is empty" "VIEWER_EMPTY"
+}
```
[the remark this answers is not in the document]
Comment (rube-de) [reply plugins/dlc/scripts/fetch-merged-pr-comments.sh:147]: Fixed: updated fetch_viewer in fetch-merged-pr-comments.sh to use // empty jq fallback and explicitly guard against null string
